ui/packages/consul-ui/tests/steps.js

```diff
@@ -103,9 +103,16 @@ export default function({
     let location = context.owner.lookup(`location:${locationType}`);
     return location.getURLFrom();
   };
+  const oidcProvider = function(name, response) {
+    const context = helpers.getContext();
+    const provider = context.owner.lookup('torii-provider:oidc-with-url');
+    provider.popup.open = async function() {
+      return response;
+    };
+  };
```

Comment:
This bit I thought would be more difficult than it actually was, I remember I'd half done this ages ago but it never went in for some reason. It allows us to mock an OIDC provider with no popup.

I first tried with popups based on work a previous PR (https://github.com/hashicorp/consul/pull/11248) which was supposed to be the road to be able to acceptance test SSO as e2e as possible, but the popup blocker gets in the way and I'm not even sure you can then interact with the popups content with Ember anyway. So I was left with this being the closest place I could mock things.